### PR TITLE
[SDESK-3286] Close popup modals with ESC key

### PR DESCRIPTION
**Pull request by marwoodandrew**

### CODE REVIEW

File: client/components/AddToPlanningModal/index.jsx [hunk 1]
```diff
@@ -8,12 +8,31 @@ import {AddToPlanningApp} from '../../apps';
...
 import {gettext} from '../../utils';
+import {KEYCODES} from '../../constants';
 
 export class AddToPlanningComponent extends React.Component {
     constructor(props) {
         super(props);
 
         this.dom = {popupContainer: null};
+        this.handleKeydown = this.handleKeydown.bind(this);
+        this.handleHide = this.props.handleHide;
```
Comment (MarkLark86) [inline client/components/AddToPlanningModal/index.jsx]: Whenever you're accessing anything from `props`, it is always better to use `this.props` everywhere in this file.
This makes it clear that you're accessing something passed down from a higher up component

File: client/components/Assignments/SelectDeskTemplate.jsx [hunk 2]
```diff
@@ -91,3 +119,5 @@ SelectDeskTemplate.propTypes = {
         privateTemplates: PropTypes.array,
     }),
 };
+
+export const SelectDeskTemplate = connect(null, null)(SelectDeskTemplateComponent);
```
Comment (MarkLark86) [inline client/components/Assignments/SelectDeskTemplate.jsx]: Using `connect` here should not be required (as `null, null` is being passed through).
You should be able to use `export class SelectDeskTemplate` instead

File: client/components/FulFilAssignmentModal/index.jsx [hunk 3]
```diff
@@ -9,79 +9,105 @@ import {Modal} from '../index';
...
+        document.removeEventListener('keydown', this.handleKeydown);
+    }
 
-        if (onCancel) {
-            onCancel();
+    handleKeydown(event) {
+        if (event.keyCode === KEYCODES.ESCAPE) {
+            event.preventDefault();
+            this.handleCancel();
+            // this.props.modalProps.$scope.resolve();
```
Comment (MarkLark86) [inline client/components/FulFilAssignmentModal/index.jsx]: Remove comment

File: client/components/NotificationModal.jsx [hunk 4]
```diff
@@ -42,3 +69,5 @@ NotificationModal.propTypes = {
         action: PropTypes.func,
     }),
 };
+
+export const NotificationModal = connect(null, null)(NotificationComponent);
```
Comment (MarkLark86) [inline client/components/NotificationModal.jsx]: Using `connect` here should not be required (as `null, null` is being passed through).
You should be able to use `export class NotificationModal` instead

Comment (marwoodandrew) [reply client/components/NotificationModal.jsx]: Done

**Pull request merged**